File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        // code here
        int maxs=0;
        int sec_max=0;
        int n=arr.size();
        if(n<2)return -1;
        for(int i=0;i<n;i++){
            if(arr[i]>maxs ){
                sec_max=maxs;
                maxs=arr[i];
            }
            else if(arr[i]>sec_max && arr[i]!=maxs){
                sec_max=arr[i];
            }
        }
        if (sec_max==0) return -1;
        return sec_max;
    }
};

File: Difficulty: Easy/Second Largest/second-largest.java
class Solution {
    public int getSecondLargest(int[] arr) {
        // code here
        int n=arr.length;
        int largest=Integer.MIN_VALUE;
        int second_largest=Integer.MIN_VALUE;
        //Arrays.sort(arr);
        for(int i=0;i<n;i++){
            if(arr[i]>largest){
                largest=arr[i];
            }
            // if(arr[i]<largest && arr[i]>second_largest){
            //     second_largest=arr[i];
            // }
        
        }
        for(int i=0;i<n;i++){
            if(arr[i]>second_largest && arr[i]<largest ){
               second_largest=arr[i]; 
            }
        }
        if(second_largest==Integer.MIN_VALUE){
            return -1;
        }
        return second_largest;
        
    }
}